samza-core/src/main/java/org/apache/samza/config/ZkConfig.java

```diff
@@ -49,7 +51,7 @@ public int getZkConnectionTimeoutMs() {
     return getInt(ZK_CONNECTION_TIMEOUT_MS, DEFAULT_CONNECTION_TIMEOUT_MS);
   }
 
-  public int getZkBarrierTimeoutMs() {
-    return getInt(ZK_CONSENSUS_TIMEOUT_MS, DEFAULT_CONSENSUS_TIMEOUT_MS);
+  public long getZkBarrierTimeoutMs() {
+    return getLong(ZK_BARRIER_TIMEOUT_MS, DEFAULT_BARRIER_TIMEOUT_MS);
```

Comment:
Hmm.. I don't remember why I changed this :P Ah ha.. And here is where you used `ZK_CONSENSUS_TIMEOUT_MS` . I will change these back.